server/src/main/java/org/elasticsearch/index/snapshots/blobstore/BlobStoreIndexShardSnapshot.java

```diff
@@ -365,24 +369,37 @@ public String toString() {
     /**
      * Constructs new shard snapshot metadata from snapshot metadata
      *
-     * @param snapshot      snapshot id
-     * @param indexVersion  index version
-     * @param indexFiles    list of files in the shard
-     * @param startTime     snapshot start time
-     * @param time          snapshot running time
-     * @param numberOfFiles number of files that where snapshotted
-     * @param totalSize     total size of all files snapshotted
+     * @param snapshot              snapshot id
+     * @param indexVersion          index version
+     * @param indexFiles            list of files in the shard
+     * @param startTime             snapshot start time
+     * @param time                  snapshot running time
+     * @param incrementalFileCount  incremental of files that were snapshotted
+     * @param incrementalSize       incremental size of snapshot
      */
-    public BlobStoreIndexShardSnapshot(String snapshot, long indexVersion, List<FileInfo> indexFiles, long startTime, long time,
-                                       int numberOfFiles, long totalSize) {
+    public BlobStoreIndexShardSnapshot(String snapshot, long indexVersion, List<FileInfo> indexFiles,
+                                       long startTime, long time,
+                                       int incrementalFileCount,
+                                       long incrementalSize
+    ) {
         assert snapshot != null;
         assert indexVersion >= 0;
         this.snapshot = snapshot;
         this.indexVersion = indexVersion;
         this.indexFiles = Collections.unmodifiableList(new ArrayList<>(indexFiles));
         this.startTime = startTime;
         this.time = time;
-        this.numberOfFiles = numberOfFiles;
+        this.incrementalFileCount = incrementalFileCount;
+        this.incrementalSize = incrementalSize;
+
+        int totalFileCount = 0;
+        long totalSize = 0;
+        for (FileInfo indexFile : indexFiles) {
+            totalFileCount ++;
```

Comment:
just `this.totalFileCount = indexFiles.size()` ? Do we even need to store this in an extra field or can we just `return indexFiles.size()` in the getter?

similarly, you can just write 

```
this.totalSize = indexFiles.stream().mapToLong(idf -> idf.metadata().length()).sum();
```